EIPS/eip-6366.md

```diff
@@ -0,0 +1,242 @@
+
+MUST trigger when permission are transferred, including zero permission transfers.
+
+A token contract which creates new tokens SHOULD emit a `Transfer` event with the `_from` address set to `address(0x00)` when tokens are created.
+
+```solidity
+event Transfer(address indexed _from, address indexed _to, uint256 _value)
+```
+
+#### Approval
+
+MUST trigger on any successful call to `approve(address _delegatee, uint256 _permission)`.
+
+```solidity
+event Approval(address indexed _owner, address indexed _delegatee, uint256 _permission)
+```
+
+### Interface EIP-6366 Metadata
+
+It is RECOMMENDED for compliant contracts to implement the optional extension `IEIP6366Meta`.
+
+```solidity
+interface IEIP6366Meta {
+  struct PermissionDescription {
+    uint256 index;
+    uint256 permission;
+    string name;
+    string description;
+  }
+
+  event UpdatePermissionDescription(uint256 indexed _permission, string indexed _name, string indexed _description);
+
+  function name() external view returns (string memory);
+
+  function symbol() external view returns (string memory);
+
+  function getDescription(uint256 _index) external view returns (PermissionDescription memory description);
+
+  function setDescription(uint256 _index, string memory _name, string memory _description) external returns (bool success);
+}
+```
+
+#### name
+
+Returns the name of the token - e.g. `"OpenPermissionToken"`.
+
+```solidity
+function name() external view returns (string memory);
+```
+
+#### symbol
+
+Returns the symbol of the token. E.g. `"OPT"`.
+
+```solidity
+function symbol() external view returns (string memory);
+```
+
+#### getDescription
+
+Return a description of a permission, at a given `_index`.
+
+```solidity
+function getDescription(uint256 _index) external view returns (PermissionDescription memory description);
+```
+
+#### setDescription
+
+Return `true` if the description was set otherwise return `false`. It MUST emit `UpdatePermissionDescription` event.
+
+```solidity
+function setDescription(uint256 _index, string memory _name, string memory _description) external returns (bool success);
+```
+
+SHOULD define a relationship between `index` field and `permission` field in `PermissionDescription`. It is RECOMMENDED to define `permission` is 2 to the power of `index`.
+
+#### UpdatePermissionDescription
+
+MUST trigger when description is updated.
+
+```solidity
+event UpdatePermissionDescription(uint256 indexed _permission, string indexed _name, string indexed _description);
+```
+
+### Interface EIP-6366 Error
+
+SHOULD NOT expected `IEIP6366Error` interface was implemented.
+
+```solidity
+interface IEIP6366Error {
+  error AccessDenied(address _owner, address _actor, uint256 _permission);
+
+  error DuplicatedPermission(uint256 _permission);
+
+  error OutOfRange();
+}
+```
+
+## Rationale
+
+Needs discussion.
+
+## Reference Implementation
+
+First implementation of [EIP-6366](https://github.com/orochi-network/EIP-6366) could be found here:
```

Comment:
Currently link to external refimpl is not allowed. Consider share core part of the implementation as a asset file or inline